be/src/connector/hive_connector.cpp

```diff
@@ -259,7 +259,25 @@ Status HiveDataSource::_init_scanner(RuntimeState* state) {
 
     HdfsScanner* scanner = nullptr;
     auto format = scan_range.file_format;
-    if (dynamic_cast<const HudiTableDescriptor*>(_tuple_desc->table_desc()) && scan_range.hudi_mor_table) {
+
+    bool use_hudi_jni_reader = false;
+    if (dynamic_cast<const HudiTableDescriptor*>(_tuple_desc->table_desc())) {
+        const auto* hudi_table = dynamic_cast<const HudiTableDescriptor*>(_hive_table);
+        bool is_mor_table = hudi_table->is_mor_table();
+        bool snapshot_mode =
+                hudi_table->get_input_format() == "org.apache.hudi.hadoop.realtime.HoodieParquetRealtimeInputFormat";
+        bool read_optimized_mode = hudi_table->get_input_format() == "org.apache.hudi.hadoop.HoodieParquetInputFormat";
+        // logs only table
+        if (is_mor_table && read_optimized_mode && scan_range.file_length == -1 && scan_range.relative_path.empty()) {
```

Comment:
Can we filter this kind of scan range in fe

Reply:
done